Feature/sva 63

sharkand73 opened the pull request:
Some hard-coded data has been added (in the form of a Javascript object and returned as an endpoint /projects.

mikemherron commented:
Looks great Andrew - much closer to the mock 👍 I'd suggest removing "lead" if we don't have a need for it yet (I can't see that displayed anywhere eon the mock). 

It may also be worth clarifying with the product owner (or whoever maintains the current Jira projects list) how the time commitments should work - I imagine that the API would want to treat this as structured data, rather than a string (i.e. `"minHoursWeek" : 4` rather than "4 hours a week") so that in future it will be searchable ("show me projects with less than 5 hours required a week") - but from the mock, I can see there are two versions show, "5-10 hours per week", and then also "6 hours" - which sounds like a one-off time commitment. If this is the case (and not just some incidental detail of the UX) then we'll need to figure out 1) how that's represented in JIRA today and 2) how we want to express it in terms of the API.

hqureshi88 commented:
Hi Andrew,

I prepared a API definition as a working document for others to work towards to (taking into account your static/hardcoded data). See below,

{ 
    "info": {
        "title": "STA Projects API",
        "version": "1.0"
    },
    "type": "object",
    "/projects": {
        "type": "array",
        "items": {
            "type": "object",
            "properties": {
                "key": {
                    "type": "string"
                },
                "name": {
                    "type": "string"
                },
                "type": {
                    "type": "string"
                },
                "client": {
                    "type": "string"
                },
                "role": {
                    "type": "string"
                },
                "description": {
                    "type": "string"
                },
                "lead": {
                    "type": "string"
                },
                "skills": {
                    "type": "string"
                },
                "hours": {
                    "type": "string"
                },
                "required": {
                    "type": "string"
                },
                "buddying": {
                    "type": "boolean"
                }
            }
        }
    }
}

This is the format we prepared so others can progress with front-end. However, feel free to comment if you think this needs some changes. I'll post in slack channel for others to see.

sharkand73 commented:
Hi Hamid,

Thanks for working on this - it’s good for me to see how these definitions work (I’ve never used API definitions before, I’ve always just ‘battered on in’!)

My PR hasn’t been merged yet, though, and I’ve had to make a couple of changes:

1)  the field “lead” has been removed at Mike’s request.  It’s not in the Figma designs (I had kept it in just in case we needed it at a later date).
2) the time commitment needs to change in format.  I’ve changed it to three fields: minHours (int), maxHours (int) and timePeriod (string) e.g. “per week”.
This may not be final.  It is my best attempt at covering all the bases, but Mike has suggested we investigate how it is stored in the Jira database.

Best,

Andrew


> On 11 Sep 2021, at 23:12, hqureshi88 ***@***.***> wrote:
> 
> 
> Hi Andrew,
> 
> I prepared a API definition as a working document for others to work towards to (taking into account your static/hardcoded data). See below,
> 
> {
> "info": {
> "title": "STA Projects API",
> "version": "1.0"
> },
> "type": "object",
> "/projects": {
> "type": "array",
> "items": {
> "type": "object",
> "properties": {
> "key": {
> "type": "string"
> },
> "name": {
> "type": "string"
> },
> "type": {
> "type": "string"
> },
> "client": {
> "type": "string"
> },
> "role": {
> "type": "string"
> },
> "description": {
> "type": "string"
> },
> "lead": {
> "type": "string"
> },
> "skills": {
> "type": "string"
> },
> "hours": {
> "type": "string"
> },
> "required": {
> "type": "string"
> },
> "buddying": {
> "type": "boolean"
> }
> }
> }
> }
> }
> 
> This is the format we prepared so others can progress with front-end. However, feel free to comment if you think this needs some changes. I'll post in slack channel for others to see.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/Scottish-Tech-Army/Volunteer-app/pull/6#issuecomment-917493036>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AS3TKMLJEPHBLG2ZA65BQCTUBPH6FANCNFSM5DVOQQFQ>.
> Triage notifications on the go with GitHub Mobile for iOS <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>. 
>

sharkand73 commented:
Hi Hamid,

I’ve just seen your post on Slack and looked at the Jira data - very exciting to finally see it!

Please ignore my change in that case - the one where I removed the “hours” field from my dummy data and replaced it with three fields.  It would appear Jira does store it as a string after all.

I think we still want to remove “lead” from the API definition.

Best,

Andrew

Sent from my iPhone

On 11 Sep 2021, at 23:53, Andrew Sharkey ***@***.***> wrote:

﻿ Hi Hamid,

Thanks for working on this - it’s good for me to see how these definitions work (I’ve never used API definitions before, I’ve always just ‘battered on in’!)

My PR hasn’t been merged yet, though, and I’ve had to make a couple of changes:

1)  the field “lead” has been removed at Mike’s request.  It’s not in the Figma designs (I had kept it in just in case we needed it at a later date).
2) the time commitment needs to change in format.  I’ve changed it to three fields: minHours (int), maxHours (int) and timePeriod (string) e.g. “per week”.
This may not be final.  It is my best attempt at covering all the bases, but Mike has suggested we investigate how it is stored in the Jira database.

Best,

Andrew


On 11 Sep 2021, at 23:12, hqureshi88 ***@***.******@***.***>> wrote:



Hi Andrew,

I prepared a API definition as a working document for others to work towards to (taking into account your static/hardcoded data). See below,

{
"info": {
"title": "STA Projects API",
"version": "1.0"
},
"type": "object",
"/projects": {
"type": "array",
"items": {
"type": "object",
"properties": {
"key": {
"type": "string"
},
"name": {
"type": "string"
},
"type": {
"type": "string"
},
"client": {
"type": "string"
},
"role": {
"type": "string"
},
"description": {
"type": "string"
},
"lead": {
"type": "string"
},
"skills": {
"type": "string"
},
"hours": {
"type": "string"
},
"required": {
"type": "string"
},
"buddying": {
"type": "boolean"
}
}
}
}
}

This is the format we prepared so others can progress with front-end. However, feel free to comment if you think this needs some changes. I'll post in slack channel for others to see.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/Scottish-Tech-Army/Volunteer-app/pull/6#issuecomment-917493036>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AS3TKMLJEPHBLG2ZA65BQCTUBPH6FANCNFSM5DVOQQFQ>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

Boopster reviewed (approved):
Looks good for merge
